filter/url_param/filterurlparam.go

```diff
@@ -0,0 +1,149 @@
+package filterurlparam
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/tsaikd/KDGoLib/errutil"
+	"github.com/tsaikd/gogstash/config"
+	"github.com/tsaikd/gogstash/config/goglog"
+	"github.com/tsaikd/gogstash/config/logevent"
+)
+
+// ModuleName is the name used in config file
+const ModuleName = "url_param"
+
+// errors
+var (
+	ErrParamParse = errutil.NewFactory("param `%q` parse error")
+)
+
+// FilterConfig holds the configuration json fields and internal objects
+type FilterConfig struct {
+	config.FilterConfig
+
+	// The field containing the user agent string.
```

Comment:
What's the relationship between `user agent` and url param filter?

Reply:
The User Agent comment should not appear here, I have modified it.